### PR TITLE
fix(dop): Solve the problem that the new project pipeline cannot be timed

### PR DESCRIPTION
**Pull request by littlejiancc**
#### What this PR does / why we need it:
Solve the problem that the new pipeline cannot be timed

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJpdGVyYXRpb25JRHMiOlsxMjgwLDEyOTddLCJhc3NpZ25lZSI6WyIxMDAxMjYxIl19&id=320557&iterationID=1280&pId=0&type=BUG)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |     Solve the problem that the new pipeline cannot be timed         |
| 🇨🇳 中文    |     解决新建流水线无法定时问题         |

#### Need cherry-pick to release versions?
/cherry-pick release/2.2

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: internal/apps/dop/providers/projectpipeline/service_impl.go [hunk 1]
```diff
@@ -415,19 +415,15 @@ func (p *ProjectPipelineService) createCronIfNotExist(definition *dpb.PipelineDe
 	if err != nil {
 		return err
 	}
+	if len(crons.Data) == 1 && crons.Data[0].PipelineDefinitionID == definition.ID {
```
Comment (sfwn) [inline internal/apps/dop/providers/projectpipeline/service_impl.go:418]: how about only one cron record but definition id is not equal?

File: internal/apps/dop/providers/projectpipeline/service_impl.go [hunk 2]
```diff
@@ -415,19 +415,15 @@ func (p *ProjectPipelineService) createCronIfNotExist(definition *dpb.PipelineDe
...
-		_, err = p.bundle.CreatePipeline(createV2)
-		if err != nil {
-			return fmt.Errorf("CreatePipeline  error %v req %v", err, createV2)
-		}
-		break
+	createV2 := extra.CreateRequest
+	createV2.DefinitionID = definition.ID
+	_, err = p.bundle.CreatePipeline(createV2)
+	if err != nil {
+		return fmt.Errorf("CreatePipeline  error %v req %v", err, createV2)
```
Comment (sfwn) [inline internal/apps/dop/providers/projectpipeline/service_impl.go:426]: format text

File: internal/apps/dop/providers/projectpipeline/service_impl_test.go [hunk 3]
```diff
@@ -904,6 +908,138 @@ func Test_makeListPipelineExecHistoryResponse(t *testing.T) {
...
...EV/feature/888/pipeline.yml","pipelineSource":"dice","labels":{"appID":"9","branch":"feature/888","commitDetail":"{\"commitID\":\"f349e5be5d2477c7de356265fc4d786202d89bdc\",\"repo\":\"https://gittar.dev.terminus.io/erda-go-demo/go-web\",\"repoAbbr\":\"erda-go-demo/go-web\",\"author\":\"dice\",\"email\":\"dice@dice.terminus.io\",\"time\":\"2022-04-20T17:56:31+08:00\",\"comment\":\"Update pipeline-yml\"}","diceWorkspace":"DEV","orgID":"1","projectID":"5"},"normalLabels":{"appName":"go-web","orgName":"erda","projectName":"go-demo"},"envs":null,"configManageNamespaces":["pipeline-secrets-app-9-default","pipeline-secrets-app-9-feature","app-9-default","app-9-dev","user-2-org-1"],"autoRun":false,"forceRun":false,"autoRunAtOnce":false,"autoStartCron":false,"cronStartFrom":null,"gc":{"resourceGC":{},"databaseGC":{"analyzed":{},"finished":{}}},"runParams":null,"definitionID":"","secrets":{"gittar.author":"dice","gittar.branch":"feature/888","gittar.commit":"f349e5be5d2477c7de356265fc4d786202d89bdc","gittar.commit.abbrev":"f349e5be","gittar.message":"Update pipeline-yml","gittar.repo":"http://gittar.project-387-dev.svc.cluster.local:5566/erda-go-demo/go-web"},"userID":"2"},"runParams":null}`
```
Comment (sfwn) [inline internal/apps/dop/providers/projectpipeline/service_impl_test.go:976]: make your test data as simple as possible

**Review by sfwn — CHANGES_REQUESTED** [review]

File: internal/apps/dop/providers/projectpipeline/service_impl.go (hunk 2 above)
Comment (littlejiancc) [reply internal/apps/dop/providers/projectpipeline/service_impl.go:426]: done

File: internal/apps/dop/providers/projectpipeline/service_impl_test.go (hunk 3 above)
Comment (littlejiancc) [reply internal/apps/dop/providers/projectpipeline/service_impl_test.go:976]: done

File: internal/apps/dop/providers/projectpipeline/service_impl.go (hunk 1 above)
Comment (littlejiancc) [reply internal/apps/dop/providers/projectpipeline/service_impl.go:418]: recreate the pipeline

**Pull request merged**